src/mon/MonMap.h

```diff
@@ -56,15 +56,17 @@ struct mon_info_t {
    * the priority of the mon, the lower value the more preferred
    */
   uint16_t priority{0};
+  uint16_t weight{10};
```

Comment:
magic number here. could you define a `constexpr` for the `DEFAULT_WEIGHT`?  and use it instead of "10".